tests/e2e/mu-plugins/e2e-rest-feature-flag.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Plugin Name: E2E Feature Flag Endpoint
+ * Description: REST Endpoint for controlling Feature Flag activation during E2E tests.
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\REST_API\REST_Routes;
+use Google\Site_Kit\Core\Util\Feature_Flags;
+
+add_action(
+	'rest_api_init',
+	function () {
+		if ( ! defined( 'GOOGLESITEKIT_PLUGIN_MAIN_FILE' ) ) {
+			return;
+		}
+
+		register_rest_route(
+			REST_Routes::REST_ROOT,
+			'e2e/feature/set-flag',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => function ( WP_REST_Request $request ) {
+					$feature_flag_overrides = get_option( 'googlesitekit_e2e_feature_flags', array() );
+
+					if ( $request['feature_value'] ) {
+						$feature_flag_overrides[ $request['feature_name'] ] = (bool) $request['feature_value'];
+					} else {
+						unset( $feature_flag_overrides[ $request['feature_name'] ] );
+					}
+
+					update_option( 'googlesitekit_e2e_feature_flags', $feature_flag_overrides );
+
+					return array(
+						'success' => true,
+					);
+				},
+				'permission_callback' => '__return_true',
+			)
+		);
+	},
+	0
+);
+
+// Enforce feature activation as defined by the E2E feature flags option.
+add_filter(
+	'googlesitekit_is_feature_enabled',
+	function( $feature_name, $feature_enabled, $mode ) {
```

Comment:
This is what's causing the E2E tests to fail, and in the WP logs I see this error:

```
[Thu Jan 21 13:57:40.095493 2021] [php7:error] [pid 166] [client 172.18.0.1:55938] PHP Fatal error: Uncaught ArgumentCountError: Too few arguments to function {closure}(), 1 passed in /var/www/html/wp-includes/class-wp-hook.php on line 289 and exactly 3 expected in /var/www/html/wp-content/mu-plugins/e2e-rest-feature-flag.php:53\nStack trace:\n#0 /var/www/html/wp-includes/class-wp-hook.php(289): {closure}(false)\n#1 /var/www/html/wp-includes/plugin.php(212): WP_Hook->apply_filters(false, Array)\n#2 /var/www/html/wp-content/plugins/google-site-kit/includes/Core/Util/Feature_Flags.php(73): apply_filters('googlesitekit_i...', false, 'serviceSetupV2', 'production')\n#3 /var/www/html/wp-content/plugins/google-site-kit/includes/Core/Util/Feature_Flags.php(87): Google\\Site_Kit\\Core\\Util\\Feature_Flags::enabled('serviceSetupV2')\n#4 /var/www/html/wp-content/plugins/google-site-kit/includes/Core/Assets/Assets.php(656): Google\\Site_Kit\\Core\\Util\\Feature_Flags::get_enabled_features()\n#5 /var/www/html/wp-content/plugins/google-site-kit/includes/Core/Assets/Assets.php(336): Google\\Site_Kit\\Core\\Assets\\Assets->get_inline_base_d in /var/www/html/wp-content/mu-plugins/e2e-rest-feature-flag.php on line 53, referer: http://localhost:9002/wp-admin/
```

Unfortunately, I can't make heads or tails of it because I'm [passing three arguments to the `apply_filters` call](https://github.com/google/site-kit-wp/blob/d979d9854f33833198aa97bef3c3e06e7e3afd4d/includes/Core/Util/Feature_Flags.php#L73) and following the docs on WP hooks from what I can see.

It claims to be passing only one argument, and indeed changing the line to:
```suggestion
	function( $feature_name ) {
```
prevents errors, but seems… wrong.

Reply:
@tofumatt the `add_Filter` call is missing the number of expected parameters here (4th argument). You only have the first 2 args here (filter handle and function). That's why it's raising the error, as me ruined previously as if not passed, apply_filters will only pass 1 argument to the callback, even if it's called with more. Confusing I know 😄

Take a look at the docs for `add_filter` and let me know if you're still stuck.

Reply:
Looking at this in the IDE now, I can see that we can simplify this since we only use the `$feature_name` anyways 😄 

We don't need to pass everything through if it isn't used. The filter is documented so we don't need to show that the other variables are available just for the sake of it.